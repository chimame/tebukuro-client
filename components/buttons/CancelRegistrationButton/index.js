// @flow
import React from 'react'

type Props = {
  eventId: ?number,
  onClick: Function,
}

export default (props: Props) => {
  const onClickHandler = (e: SyntheticEvent<>) => {
    e.preventDefault()
    props.onClick({ eventId: props.eventId })
  }

  return (
    <button onClick={onClickHandler}>
      登録をキャンセルする
    </button>
  )
}
