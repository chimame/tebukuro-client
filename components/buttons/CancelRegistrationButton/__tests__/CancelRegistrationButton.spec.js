import React from 'react'
import { shallow, mount } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import CancelRegistrationButton from '../index'

const testProps = {
  eventId: 1,
}

describe('CancelRegistrationButton', () => {
  describe('with event has a waitlist', () => {
    it('renders CancelRegistrationButton.', () => {
      const wrapper = shallow(<CancelRegistrationButton />)
      const tree = shallowToJson(wrapper)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when clicked', () => {
    const onCancel = jest.fn()
    const wrapper =
      mount(<CancelRegistrationButton {...testProps} onClick={onCancel} />)

    wrapper.find('button').simulate('click')

    it('should call onCancel once.', () => {
      expect(onCancel).toHaveBeenCalledTimes(1)
    })

    it('should call onCancel with correct argument.', () => {
      expect(onCancel).toBeCalledWith({
        eventId: testProps.eventId,
      })
    })
  })
})
