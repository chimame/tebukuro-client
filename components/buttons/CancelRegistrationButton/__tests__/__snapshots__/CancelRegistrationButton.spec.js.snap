// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelRegistrationButton with event has a waitlist renders CancelRegistrationButton. 1`] = `
<button
  onClick={[Function]}
>
  登録をキャンセルする
</button>
`;
